packages/jest-phabricator-coverage/src/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2014, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ * @flow
+ */
+
+'use strict';
+
+import type {
+  AggregatedResult,
+  CoverageMap,
+} from 'types/TestResult';
+
+type PhabricatorReport = {
+  phabricatorReport: Array<Object>
+};
+
+const {formatTestResults} = require('jest-util');
+
+function summarize(coverageMap: CoverageMap, filterBy: Array<string>) {
+  let summaries = Object.create(null);
+
+  coverageMap.files().forEach(file => {
+    const covered = [];
+    const lineCoverage = coverageMap.fileCoverageFor(file).getLineCoverage();
+
+    Object.keys(lineCoverage).forEach(lineNumber => {
+      // Line numbers start at one
+      const number = (parseInt(lineNumber, 10) - 1);
+      const visited = !!lineCoverage[lineNumber];
+      covered[number] = visited ? 'C' : 'U';
+    });
+
+    for (let i = 0; i < covered.length; i++) {
+      if (!covered[i]) {
+        covered[i] = 'N';
+      }
+    }
+
+    summaries[file] = covered.join('');
+  });
+
+  if (filterBy.length) {
+    summaries = filterBy.reduce((result, file) => {
+      result[file] = summaries[file];
+      return result;
+    }, {});
```

Comment:
Do we really need a reducer here? I think it's more straightforward to do:

```
const results = Object.create(null);
filterBy.forEach(file => result[file] = summaries[file]);
summaries = result;
```

Also note that we want to create a prototype-less object here.

Although, thinking about this more, I think we can use a different approach here that is going to be more efficient. In this current implementation we are collecting coverage for everything and then filtering the ones we care about. It seems to me like we can normalize the filter object before and then simply skipping the files we don't care about above. That way we'll only do the minimum amount of work that is necessary.